web-runner.go

```diff
@@ -0,0 +1,237 @@
+			_ = killProcess(pid)
+
+			// Cleanup temp folder
+			err = os.RemoveAll(flags.RunInfoFolder)
+			if err != nil {
+				log.Debug(err.Error())
+			}
+
+			return errors.New("couldn't start background process")
+		case <-ticker.C:
+			runInfo, _ := server.GetRunInfo(flags.RunInfoFolder)
+			if runInfo == nil {
+				continue
+			}
+
+			cli.Println("Web running in background with PID:%d", pid)
+			if runInfo.BaseURL != "" {
+				cli.Println(runInfo.BaseURL)
+			}
+
+			return nil
+		}
+	}
+}
+
+func runBackgroundCmd(cmd *cobra.Command, pidFile string) (int, error) {
+	err := os.MkdirAll(filepath.Dir(pidFile), 0750)
+	if err != nil {
+		return 0, fmt.Errorf("not possible to create tmp directory for %s", pidFile)
+	}
+
+	// Prepare the command to restart itself in the background
+	args := append([]string{cmd.Name()}, os.Args[2:]...)
+
+	command := exec.Command(os.Args[0], args...) //nolint G204
+	command.Env = append(os.Environ(), fmt.Sprintf("%s=1", backgroundCmdEnvVar))
+
+	// Set platform-specific process ID
+	setSysProcAttr(command)
+
+	err = command.Start()
+	if err != nil {
+		cmd.Println("Failed to start in background:", err)
+		return 0, err
+	}
+
+	err = os.WriteFile(pidFile, []byte(fmt.Sprintf("%d", command.Process.Pid)), os.FileMode(0644))
+	if err != nil {
+		cmd.Println("Failed to write PID file:", err)
+		return 0, err
+	}
+
+	return command.Process.Pid, nil
+}
+
+func redirectOutputs(folder string) error {
+	// @TODO rework logs, to replace with global launchr logging.
+	err := os.MkdirAll(folder, 0750)
+	if err != nil {
+		return fmt.Errorf("can't create plugin temporary directory")
+	}
+
+	outLog, err := os.Create(fmt.Sprintf("%s/out.log", folder))
+	if err != nil {
+		return err
+	}
+
+	errLog, err := os.Create(fmt.Sprintf("%s/error.log", folder))
+	if err != nil {
+		return err
+	}
+
+	os.Stdout = outLog
+	os.Stderr = errLog
+
+	return nil
+}
+
+func isWebRunning(pidFile string, runInfoDir string) (bool, string) {
+	url := ""
+	if isBackGroundEnv() {
+		return false, url
+	}
+
+	serverRunInfo, err := server.GetRunInfo(runInfoDir)
+	if err != nil {
+		log.Debug(err.Error())
+		return false, url
+	}
+
+	if serverRunInfo == nil || serverRunInfo.BaseURL == "" {
+		url = serverRunInfo.BaseURL
+	}
+
+	if pidFileExists(pidFile) {
+		pid, errPid := readPidFile(pidFile)
+		if errPid == nil {
+			if isProcessRunning(pid) {
+				return true, url
+			}
+		}
+	}
+
+	if url == "" {
+		return false, url
+	}
+
+	isHealthy, err := server.CheckHealth(serverRunInfo.BaseURL)
+	if err != nil {
+		log.Debug(err.Error())
+	}
+
+	return isHealthy, url
+}
+
+func stopWeb(pidFile, runInfoDir string) error {
+	onSuccess := "The web server has been successfully shut down."
+
+	if pidFileExists(pidFile) {
+		pid, err := readPidFile(pidFile)
+		if err == nil {
+			if isProcessRunning(pid) {
+				err = interruptProcess(pid)
+				if err != nil {
+					log.Debug(err.Error())
+				}
+				return nil
+			}
+		} else {
+			log.Debug(err.Error())
```

Comment:
It is really strange to `log.debug` an error. If you don't expect an error, you must panic or return an error. Soft logging a PID error is not ok because it's the core of your background functionality. Considering that, reorganize ifs to return early.